### PR TITLE
Bfe 35 berps delete the markets page to replace a markets dropdown

### PR DESCRIPTION
**Pull request by bearpong**
Removes `/markets` page and restyles create position component

### CODE REVIEW

**Comment by BrownBrownBear** [comment]
1. From a design perspective, can we move the markets table from the old markets page to the home page? I think it has some important information and can go right below the markets infinite scroll tiles. 
2. Can we add open interest in both long and short to the new markets drop down? It doesn't have to be green/red and can be just a flat white.
3. This doesn't have to be added now, but we'll need to eventually put a max height on the dropdown and add a scroll when we have more markets. Just something to be wary about. 
4. I've added a screen recording here. I feel like if we're minimizing the TP/SL/Leverage, we should always have the defaults set in place when we switch the type of order (Long/Short,Market/Limit). This provides a cleaner UX as the user doesn't need to open the advanced details and their order will go through with "expected configuration".

https://github.com/user-attachments/assets/4979beff-90bc-4478-a949-5f1a5f860089

**Review by BrownBrownBear — CHANGES_REQUESTED** [review]
Code looks good to me. JUst need those changes from my previous comment

**Comment by BrownBrownBear** [comment]
> Sorry, I had forgot about the open interest updates, done now. About the market table in the homepage, I agree there's some useful information but it'd be very similiar to the data in the markets infinite scroll tiles, I'd avoid repetitions. Maybe should we consider it for the homepage redesign bobby's working onto?

Sounds good. We can keep it as is and don't worry about adding that table. we can look back into it when we decide to go live with mainnet for perps. 

Three last quick changes for perps:
- `Open Interest` header to `Open Interest (L/S)`
- Add 24H trades for each market to market details popover
- Fix padding on dropdown when in smaller screens

**Comment by BrownBrownBear** [comment]
Found a bug when switching between long and short, the long buttons remain grayed out:

https://github.com/user-attachments/assets/6c96ec01-1753-45d9-8d1e-0501b61b853d

File: apps/perp/src/app/berpetuals/components/instrument-dropdown.tsx [hunk 1]
```diff
@@ -134,53 +255,19 @@ export function InstrumentDropdown({
           )}
         </div>
       </DropdownMenuTrigger>
+
       <DropdownMenuContent
-        className={`mx-2 my-1 flex w-[calc(100vw-16px)] flex-col rounded-md border border-border bg-background p-0 lg:w-[400px] ${DYNAMIC_DROPDOWN_MOBILE_HEIGHTS[activeBanners]} ${DYNAMIC_DROPDOWN_DESKTOP_HEIGHTS[activeBanners]}`}
+        className={`my-1 w-full p-0 max-lg:min-w-[calc(100vw-1rem)] lg:mx-2 ${DYNAMIC_DROPDOWN_MOBILE_HEIGHTS[activeBanners]} ${DYNAMIC_DROPDOWN_DESKTOP_HEIGHTS[activeBanners]}`}
```
Comment (BrownBrownBear) [inline apps/perp/src/app/berpetuals/components/instrument-dropdown.tsx:260]: Here's the padding issue where the left side is padding is not aligned with the right side
Screenshot 2024-08-30 at 12 24 18 PM

File: apps/perp/src/app/berpetuals/components/instrument-dropdown.tsx [hunk 2]
```diff
@@ -83,46 +86,164 @@ const MarketPriceOverview = ({
...
+    accessorKey: "dailyHistoricPrice",
+    enableSorting: false,
+    size: 150,
+    // minSize: 130,
+    // meta: {
+    //   className: "max-md:hidden",
+    // },
+  },
+  {
+    header: "Open Interest",
```
Comment (BrownBrownBear) [inline apps/perp/src/app/berpetuals/components/instrument-dropdown.tsx:143]: Can you change this to: `Open Interest (L/S)`

File: apps/perp/src/app/berpetuals/components/tpsl.tsx [hunk 3]
```diff
@@ -116,352 +172,157 @@ export function TPSL({
...
+                ? "(Unset)"
+                : `${
+                    realTpPercent
+                      ? `(${
+                          tpPercentBN.lt("0")
+                            ? "Less than "
+                            : tpPercentBN.gt(MAX_TP_PERC)
+                              ? "Greater than "
+                              : ""
+                        }${tpPercentBN.toString(10)}%)`
```
Comment (BrownBrownBear) [inline apps/perp/src/app/berpetuals/components/tpsl.tsx:260]: Can we try and clean this percentage up when it's an invalid profit (i.e. placing a long order and we set a take profit lower than the mark price of the market)
Screenshot 2024-08-30 at 12 39 29 PM
Screenshot 2024-08-30 at 12 39 32 PM

Comment (BrownBrownBear) [reply apps/perp/src/app/berpetuals/components/tpsl.tsx:260]: I've added the before and after

**Comment by BrownBrownBear** [comment]
Another weird interaction. I understand what's happening but was wondering if we could make the UX better. The max SL is 300% and that's what it's currently is but the SL button that's highlight is the grayed out 900%. Not sure what the best way to resolve this is but maybe we can unhighlight 900% when it's not available and if the TP/SL price lines up with the percentage options we give the user (50, 100, 300, 900), they'll be highlighted? 


https://github.com/user-attachments/assets/ef45bd3e-8089-461e-80ab-c8ca9ad3ac99

File: packages/shared-ui/src/approve-button.tsx [hunk 4]
```diff
@@ -49,28 +49,11 @@ export const ApproveButton = ({
     },
   });
   const ref = useRef<HTMLDivElement>(null);
-  const [smallScreen, setSmallScreen] = useState(false);
-
-  useEffect(() => {
```
Comment (BrownBrownBear) [inline packages/shared-ui/src/approve-button.tsx:54]: Good use of removing calculating client width here!

File: packages/berajs/src/hooks/useTpsl.ts [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+        assetPrice === "0" ||
+        assetPrice === "" ||
+        leverage === "0" ||
+        leverage === ""
+      ) {
+        setMaxPercentage("");
+      }
+      const perc = getPercentageFromPrice(
+        long,
+        "0.00000001",
```
Comment (BrownBrownBear) [inline packages/berajs/src/hooks/useTpsl.ts:80]: Why is this 0.00000001?

**Review by BrownBrownBear — CHANGES_REQUESTED** [review]

**Comment by bearpong** [comment]
Yup, i noticed that, however i think this is the best way so far and gns behaves in the same way. If user clicks on 900%, i don't think it'd be good UX pattern if we unselected 900% as the leverage goes down, even for a second. what do you say?

File: packages/berajs/src/hooks/useTpsl.ts (hunk 5 above)
Comment (bearpong) [reply packages/berajs/src/hooks/useTpsl.ts:80]: nvm i'm setting it to 0

File: packages/shared-ui/src/approve-button.tsx (hunk 4 above)
Comment (bearpong) [reply packages/shared-ui/src/approve-button.tsx:54]: thank youu

File: apps/perp/src/app/berpetuals/components/tpsl.tsx (hunk 3 above)
Comment (bearpong) [reply apps/perp/src/app/berpetuals/components/tpsl.tsx]: ty, fixed

**Pull request merged**